src/com/java/multithreading/BlockingQueue.java

```diff
@@ -0,0 +1,46 @@
+package com.java.multithreading;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class BlockingQueue {
+
+  private Queue<Integer> queue;
+  private int capacity;
+
+  public BlockingQueue(int capacity) {
+    this.queue = new LinkedList<Integer>();
+    this.capacity = capacity;
+  }
+
+  public synchronized  void enqueue(Integer item){
```

Comment:
Instead of making whole method synchronized, can you try limiting the scope of syncrhonized?